packages/libs/lambda-at-edge/src/default-handler.ts

```diff
@@ -47,14 +47,22 @@ const addS3HostHeader = (
 const isDataRequest = (uri: string): boolean => uri.startsWith("/_next/data");
 
 const normaliseUri = (uri: string): string => {
-  if (basePath) uri = uri.slice(basePath.length);
+  if (basePath) {
+    if (uri.startsWith(basePath)) {
+      uri = uri.slice(basePath.length);
+    } else {
+      // basePath set but URI does not start with basePath, return 404
+      return "/404";
```

Comment:
This is a reserved route that will let us serve the 404.html (static page) (generated by Next.js), won't conflict with any user pages.